contracts/multicall/GuardedMulticaller2.sol

```diff
@@ -0,0 +1,240 @@
+// Copyright Immutable Pty Ltd 2018 - 2023
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+// Signature Validation
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+
+// Access Control
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+
+// Reentrancy Guard
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+// EIP-712 Typed Structs
+import {EIP712} from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+
+/**
+ *
+ * @title GuardedMulticaller2 contract
+ * @author Immutable
+ * @notice This contract is used to batch calls to other contracts.
+ * @dev This contract is not designed to be upgradeable. If an issue is found with this contract,
+ *  a new version will be deployed. All approvals granted to this contract will be revoked before
+ *  a new version is deployed. Approvals will be granted to the new contract.
+ */
+contract GuardedMulticaller2 is AccessControl, ReentrancyGuard, EIP712 {
+    /// @dev Struct for call data
+    struct Call {
+        address target;
+        string functionSignature;
+        bytes data;
+    }
+
+    /// @dev Mapping of reference to executed status
+    // solhint-disable-next-line named-parameters-mapping
```

Comment:
Is there a reason for not removing the solhint disable, and adding the name of the parameters in the mapping?